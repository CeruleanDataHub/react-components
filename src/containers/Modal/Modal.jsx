import PropTypes from "prop-types";
import React from "react";
import ModalWrapper from "styled-react-modal";

const StyledModal = ModalWrapper.styled`
  width: 20rem;
  height: 20rem;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: white;
`;

export const Modal = ({ children, isOpen, onBackgroundClick }) => (
  <StyledModal isOpen={isOpen} onBackgroundClick={onBackgroundClick}>
    {children}
  </StyledModal>
);

Modal.propTypes = {
  /** Content of the modal */
  children: PropTypes.node,
  /** Is modal open or not */
  isOpen: PropTypes.bool,
  /** Handler function */
  onBackgroundClick: PropTypes.func
};

Modal.defaultProps = {
  children: "",
  isOpen: false,
  onBackgroundClick: () => null
};
