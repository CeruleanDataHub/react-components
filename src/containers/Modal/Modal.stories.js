import React, { useState } from "react";

import { Button } from "../Button/Button";
import { Typography } from "../Typography/Typography";
import { Modal } from "./Modal";

export default {
  title: "Modal",
  component: Modal
};

const ModalToggle = () => {
  const [isOpen, setIsOpen] = useState(false);

  const toggleModal = () => {
    setIsOpen(!isOpen);
  };

  return (
    <Typography fontFamily="openSans">
      <Button onClick={toggleModal} color="blue">
        Open modal
      </Button>
      <Modal isOpen={isOpen} onBackgroundClick={toggleModal}>
        <p>Modal content</p>
      </Modal>
    </Typography>
  );
};

export const ModalToggleStory = () => (
  <div style={{ minHeight: 500 }}>
    <ModalToggle />
  </div>
);
