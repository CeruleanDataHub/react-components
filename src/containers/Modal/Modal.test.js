import React from "react";
import renderer from "react-test-renderer";

import { Modal } from "./Modal";

describe("Modal", () => {
  it("should render open Modal with children", () => {
    const component = renderer.create(
      <Modal isOpen>
        <p>Modal content</p>
      </Modal>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call default props onBackgroundClick function when no onBackgroundClick property is passed", () => {
    const component = renderer.create(
      <Modal isopen>
        <p>Modal content</p>
      </Modal>
    );
    component.root.props.onBackgroundClick();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
