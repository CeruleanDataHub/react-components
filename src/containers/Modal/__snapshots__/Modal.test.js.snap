// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should call default props onBackgroundClick function when no onBackgroundClick property is passed 1`] = `null`;

exports[`Modal should render open Modal with children 1`] = `null`;
